packages/across-monitor/src/RelayEventProcessor.ts

```diff
@@ -58,58 +59,24 @@ export class RelayEventProcessor {
       this.deposits[l1TokenAddress] = this.deposits[l1TokenAddress] ? this.deposits[l1TokenAddress] : {};
     }
 
-    await Promise.all(
-      Object.keys(this.bridgePools).map(async (l1TokenAddress) => {
-        const depositRelayedEvents = await this.bridgePools[l1TokenAddress].contract.getPastEvents("DepositRelayed", {
-          fromBlock: startingBlock,
-          toBlock: endingBlock,
-        });
-        for (const depositRelayedEvent of depositRelayedEvents) {
-          const relayData: Relay = {
-            chainId: Number(depositRelayedEvent.returnValues.depositData.chainId),
-            depositId: Number(depositRelayedEvent.returnValues.depositData.depositId),
-            l2Sender: depositRelayedEvent.returnValues.depositData.l2Sender,
-            l1Recipient: depositRelayedEvent.returnValues.depositData.l1Recipient,
-            amount: depositRelayedEvent.returnValues.depositData.amount,
-            slowRelayFeePct: depositRelayedEvent.returnValues.depositData.slowRelayFeePct,
-            instantRelayFeePct: depositRelayedEvent.returnValues.depositData.instantRelayFeePct,
-            realizedLpFeePct: depositRelayedEvent.returnValues.relay.realizedLpFeePct,
-            depositHash: depositRelayedEvent.returnValues.depositHash,
-          };
-          this.deposits[l1TokenAddress][depositRelayedEvent.returnValues.depositHash] = relayData;
-        }
-      })
-    );
+    for (const relayData of this.l1Client.getAllRelayedDeposits()) {
```

Comment:
`l1Client` deletes pending relays if they are disputed, hence `getRelayEventInfo` could throw as `depositHash` emitted on dispute could not be matched to Relay data.

Reply:
yes, this would indeed help. and assume it would store all deposits from the earliest block